### PR TITLE
Secured app, add error pages

### PR DESCRIPTION
**Pull request by AndLvovSky**
Used flyway instead of CommandLineRunner for loading data for production; added custom login page; created User, Role, Privilege models, UserRepository and UserDetailsService; added custom error controller; used BCryptPasswordEncoder.

### CODE REVIEW

File: src/main/java/com/andlvovsky/periodicals/model/publication/Publication.java [hunk 1]
```diff
@@ -3,18 +3,16 @@
...
 
 @Data
 @Entity
 @NoArgsConstructor
 public class Publication {
 
     @Id
-    @GeneratedValue(strategy=GenerationType.AUTO)
+    @GeneratedValue(strategy=GenerationType.SEQUENCE, generator="publication_generator")
+    @SequenceGenerator(name="publication_generator", sequenceName="publication_sequence", allocationSize=1)
```
Comment (igor-dmitriev) [inline src/main/java/com/andlvovsky/periodicals/model/publication/Publication.java]: formatting

Comment (AndLvovSky) [reply src/main/java/com/andlvovsky/periodicals/model/publication/Publication.java]: I thouht it would have better readability if we don't put spaces around equal sign like in python with named arguments

File: src/main/java/com/andlvovsky/periodicals/config/WebSecurityConfig.java [hunk 2]
```diff
@@ -37,7 +37,7 @@ protected void configure(HttpSecurity http) throws Exception {
 
     @Bean
     public PasswordEncoder passwordEncoder() {
-        return new BCryptPasswordEncoder();
```
Comment (igor-dmitriev) [inline src/main/java/com/andlvovsky/periodicals/config/WebSecurityConfig.java]: Why did you remove it? BcryptPasswordEncoder is a good way to encode password

File: src/main/resources/db/migration/V1__Initialization.sql [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+);
+
+CREATE TABLE user_entity_roles (
+    users_id bigint NOT NULL,
+    roles_id bigint NOT NULL,
+    FOREIGN KEY (roles_id) REFERENCES role(id),
+    FOREIGN KEY (users_id) REFERENCES user_entity(id)
+);
+
+INSERT INTO privilege (id, name) VALUES
```
Comment (igor-dmitriev) [inline src/main/resources/db/migration/V1__Initialization.sql]: I recommend to split ddl and dml operations

File: src/main/java/com/andlvovsky/periodicals/config/WebSecurityConfig.java (hunk 2 above)
Comment (AndLvovSky) [reply src/main/java/com/andlvovsky/periodicals/config/WebSecurityConfig.java]: I think, it's optional for tests to encrypt passwords. So, I put it only in the production.
